plugins/modules/purefa_volume.py

```diff
@@ -456,101 +458,117 @@ def get_multi_volumes(module, destroyed=False):
             + str(vol_num).zfill(module.params["digits"])
             + module.params["suffix"]
         )
-    return bool(array.get_volumes(names=names, destroyed=destroyed).status_code == 200)
+    if LooseVersion(CONTEXT_API_VERSION) <= LooseVersion(api_version):
+        res = array.get_volumes(
+            names=names, destroyed=False, context_names=[module.params["context"]]
+        )
+    else:
+        res = array.get_volumes(names=names, destroyed=False)
+    if res.status_code == 200:
+        return list(res.items)[0]
+    return None
 
 
 def get_volume(module, array):
     """Return Volume or None"""
-    try:
-        return array.get_volume(module.params["name"])
-    except Exception:
-        return None
+    api_version = array.get_rest_version()
+    if LooseVersion(CONTEXT_API_VERSION) <= LooseVersion(api_version):
+        res = array.get_volumes(
+            names=[module.params["name"]], context_names=[module.params["context"]]
+        )
+    else:
+        res = array.get_volumes(names=[module.params["name"]])
+    if res.status_code == 200:
+        return list(res.items)[0]
+    return None
 
 
-def get_endpoint(name, array):
+def get_endpoint(module, name, array):
     """Return Endpoint or None"""
-    try:
-        return array.get_volume(name, pending=True, protocol_endpoint=True)
-    except Exception:
-        return None
+    api_version = array.get_rest_version()
+    if LooseVersion(CONTEXT_API_VERSION) <= LooseVersion(api_version):
+        res = array.get_volumes(names=[name], context_names=[module.params["context"]])
+    else:
+        res = array.get_volumes(names=[name])
+    if res.status_code == 200:
+        volume = list(res.items)[0]
+        if volume.protocol_endpoint:
+            return list(res.items)[0]
```

Comment:
Already iterator output has been consumed in line 494, this leads to index out of error, please recheck on this.